tools/depcheck/pkg/cmd/pkg.go

```diff
@@ -108,6 +118,43 @@ func BuildGraph(packages *PackageList, excludes []string) (*depgraph.MutableDire
 	return g, nil
 }
 
+// CalculatePackages receives a set of go import paths and runs "go list"
+// using each path as an entrypoint.
+// Returns a PackageList containing dependency and importPath data for each package.
+func CalculatePackages(entrypoints []string) (*PackageList, error) {
+	args := []string{"list", "--json"}
+	golist := exec.Command("go", append(args, entrypoints...)...)
+
+	r, w := io.Pipe()
+	golist.Stdout = w
+	golist.Stderr = os.Stderr
+
+	pkgs := &PackageList{}
+	go func(list *PackageList) {
+		decoder := json.NewDecoder(r)
+		for {
+			var pkg Package
+			err := decoder.Decode(&pkg)
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "error: %v\n", err)
+			}
+
+			list.Add(pkg)
+		}
+	}(pkgs)
+
+	// no need to wait for goroutine to finish, golist.Run() will
+	// block until the command finishes executing.
```

Comment:
Doesn't seem likely.  Wouldn't it wait until the buffer was consumed, not until the list.Add call finished?